src/Microsoft.Sbom.Api/Output/Telemetry/TelemetryRecorder.cs

```diff
@@ -46,6 +47,66 @@ public TelemetryRecorder(IFileSystemUtils fileSystemUtils, IConfiguration config
             Log = log ?? throw new ArgumentNullException(nameof(log));
         }
 
+        private TelemetryRecorder(string telemetryFilePath, IFileSystemUtils fileSystemUtils)
+        {
+            this.telemetryFilePath = telemetryFilePath;
+            FileSystemUtils = fileSystemUtils ?? throw new ArgumentNullException(nameof(fileSystemUtils));
+        }
+
+        public static TelemetryRecorder Create(string telemetryFilePath, IFileSystemUtils fileSystemUtils)
+        {
+            return new TelemetryRecorder(telemetryFilePath, fileSystemUtils);
+        }
+
+        //Method to log telemetry in conditions when the tool is not able to start execution of workflow.
+        public async Task ConsoleLogger(Exception e)
+        {
+            var logger = new LoggerConfiguration().WriteTo.Console().CreateLogger();
+            try
+            {
+                //convert exception to IDictionary<string,string> and handle unicode char.
+                var exceptions = new Dictionary<string, string>
+                {
+                    { e.GetType().ToString().Replace("\u0027", "'"), e.Message }
+                };
+
+                // Create the telemetry object.
+                var telemetry = new SBOMTelemetry
+                {
+                    Result = Result.Failure,
+                    Timings = timingRecorders.Select(t => t.ToTiming()).ToList(),
+                    Switches = this.switches,
+                    Exceptions = exceptions
+                };
+
+                // Log to logger.
+                logger.Information("Could not start execution of workflow. Logging telemetry {@Telemetry}", telemetry);
```

Comment:
If this logger is the one being called it is because it couldn't reach the loggers present in the workflows and we are throwing an exception, so I believe it is always relevant.